tekton/build_pipeline.go

```diff
@@ -45,6 +45,11 @@ func AnnotateBuildPipelineRun(ctx context.Context, pipelineRun *tektonv1.Pipelin
 func AnnotateBuildPipelineRunWithCreateSnapshotAnnotation(ctx context.Context, pipelineRun *tektonv1.PipelineRun, cl client.Client, ensureSnapshotExistsErr error) error {
 	message := ""
 	status := ""
+
+	if pipelineRun.GetDeletionTimestamp() != nil {
```

Comment:
Shouldn't be this logic at different place? When pipeline is being deleted we shouldn't even call functions that edit data.

Or is it here because of the RetryOnConflict usage?

Reply:
Yeah, I agree with you, was thinking to streamline the retryOnConflict, but it makes more sense to be checked before trying to annotate the build PLR.